pyiceberg/table/__init__.py

```diff
@@ -621,6 +621,13 @@ def delete(self, delete_filter: Union[str, BooleanExpression], snapshot_properti
         if not delete_snapshot.files_affected and not delete_snapshot.rewrites_needed:
             warnings.warn("Delete operation did not match any records")
 
+    def find_referenced_files(self, file_paths: List[str]) -> list[str]:
```

Comment:
We want to be intentional about introducing public methods. Since this is just used by `add_files`, could we make this method private? The type notation of `list` in the return type also needs to be `typing.List` until we deprecate python3.8 support:

```suggestion
    def _find_referenced_files(self, file_paths: List[str]) -> List[str]:
```

An alternative is to just keep this logic within `add_files`, since this method isn't reused and is just 2 or 3 lines of code

Reply:
you are right, it makes more sense to move it into the logic into add_files since no one uses it and its on 3 lines if code